yt_dlp/extractor/cbc.py

```diff
@@ -490,32 +492,32 @@ class CBCGemLiveIE(InfoExtractor):
         'skip': 'Live might have ended',
     }
 
-    # It's unclear where the chars at the end come from, but they appear to be
-    # constant. Might need updating in the future.
-    # There are two URLs, some livestreams are in one, and some
-    # in the other. The JSON schema is the same for both.
-    _API_URLS = ['https://tpfeed.cbc.ca/f/ExhSPC/t_t3UKJR6MAT', 'https://tpfeed.cbc.ca/f/ExhSPC/FNiv9xQx_BnT']
-
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        video_info = json.loads(get_element_by_id('__NEXT_DATA__', webpage))['props']['pageProps']['data']
 
-        for api_url in self._API_URLS:
+        # Two types of metadata JSON
+        if not video_info.get('formattedIdMedia'):
             video_info = next((
-                stream for stream in self._download_json(api_url, video_id)['entries']
-                if stream.get('guid') == video_id), None)
-            if video_info:
-                break
-        else:
+                item for item in video_info['freeTv']['items']
+                if item.get('key') == video_id), None)
+
+        video_stream_id = video_info.get('formattedIdMedia')
+        if not video_stream_id:
             raise ExtractorError('Couldn\'t find video metadata, maybe this livestream is now offline', expected=True)
 
+        stream_data = self._download_json(
+            f'https://services.radio-canada.ca/media/validation/v2/?appCode=mpx&connectionType=hd&deviceType=ipad&idMedia={video_stream_id}&multibitrate=true&output=json&tech=hls&manifestType=desktop',
+            video_id)
+
         return {
             '_type': 'url_transparent',
-            'ie_key': 'ThePlatform',
-            'url': video_info['content'][0]['url'],
-            'id': video_id,
+            'url': stream_data['url'],
+            'ie_key': 'Generic',
```

Comment:
why return a `url_transparent` info_dict to `Generic`?
we can extract m3u8 formats in this extractor, yeah?

`traverse_obj` can also be used to make this cleaner:
```py
        return {
            'id': video_id,
            'formats': self._extract_m3u8_formats(stream_data['url'], video_id, 'mp4', live=True),
            'is_live': True,
            **traverse_obj(video_info, {
                'title': 'title',
                'description': 'description',
                'thumbnail': ('images', 'card', 'url'),
                'timestamp': ('airDate', {parse_iso8601}),
            })
        }
```